### PR TITLE
Feature/httpx abstractions

### PR DESCRIPTION
**Pull request by lxmnk**

### CODE REVIEW

File: botx/clients/clients/sync_client.py [hunk 1]
```diff
@@ -29,27 +29,34 @@ def __post_init__(self) -> None:
...
+
+        Arguments:
+            method: BotX method.
+
+        Returns:
+            Built request.
+        """
+        return method.build_http_request()
+
+    def process_response(
```
Comment (Opti213) [inline botx/clients/clients/sync_client.py:44]: `execute`?

File: botx/clients/clients/sync_client.py [hunk 2]
```diff
@@ -59,35 +66,37 @@ def call(
...
                 status=response.status_code,
-                response_content=response.json(),
+                response_content=response.json_body,
             )
 
         return extract_result(method, response)
 
-    def execute(self, method: BotXMethod) -> Response:
-        """Make real HTTP request using client.
+    def execute(self, method: BotXMethod, request: HTTPRequest) -> HTTPResponse:
```
Comment (Opti213) [inline botx/clients/clients/sync_client.py:74]: `call`?

File: botx/clients/methods/wrappers.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    """Wrapper for HTTP response."""
+
+    #: response status code.
+    status_code: int
+
+    #: content
+    bytes_body: bytes
+
+    #: response json.
+    json_body: Dict[str, Any]
```
Comment (Opti213) [inline botx/clients/methods/wrappers.py:38]: i think we should use conventional interface and naming. https://github.com/psf/requests/blob/master/requests/models.py#L589

File: botx/models/constants.py [hunk 4]
```diff
@@ -1,3 +1,11 @@
...
 
 MAXIMUM_TEXT_LENGTH = 4096
+
+
+class HTTPCodes:
+    BAD_REQUEST = 400
+    FORBIDDEN = 403
+    NOT_FOUND = 404
+    IM_A_TEAPOT = 418
+    UNPROCESSABLE_ENTITY = 422
```
Comment (Opti213) [inline botx/models/constants.py:11]: Why not use `http` module?

File: botx/clients/methods/wrappers.py [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+
+    @property
+    def is_successful(self) -> bool:
```
Comment (Opti213) [inline botx/clients/methods/wrappers.py:41]: nit: success - its usual result, maybe better use `is_error`?

Comment (lxmnk) [reply botx/clients/methods/wrappers.py:41]: We expects only 2xx, no redirects, no client or server errors, so I think `is_success` is ok.

File: botx/clients/methods/wrappers.py (hunk 3 above)
Comment (lxmnk) [reply botx/clients/methods/wrappers.py:38]: `json` is reserved name in pydantic. Also, we point that it's not requests/httpx response.

File: botx/models/constants.py (hunk 4 above)
Comment (lxmnk) [reply botx/models/constants.py:11]: Good idea, thank you!

**Pull request closed without merging**